centinel/primitives/dnslib.py

```diff
@@ -94,12 +94,8 @@ def lookup_domains(self):
         """
         for nameserver in self.nameservers:
             for domain in self.domains:
-                escape_count = 0
                 while threading.active_count() > self.max_threads:
                     time.sleep(1)
-                    escape_count += 1
-                    if escape_count > 100:
```

Comment:
that's true. It was supposed to be a break, not a continue. Either way, I think it would be better to refactor with a maximum timeout, then raise an exception if that timeout is exceeded.